### DISCUSSION TITLE
Is 99.999999998 possible to return true using chance.bool?

### DISCUSSION THREAD

**Discussion by Darren120** [chancejs/chancejs#579] [Q&A] [answered]
title. or does it round it?

**Comment by victorquinn** [comment]
I'm sorry I don't really understand your question.

But you could do the following:
```js
chance.bool({likelihood: 99.999999998})
```

to get a true value from Chance 99.999999998% of the time

So I think the answer to your question is yes, it is possible to return `true` with that possibility, but I am not sure if I understand your question.

**Comment by somejeff** [comment, marked as answer]
Hi @Darren120,
To be overly specific, when it comes to floats, JavaScript uses double precision floating point numbers. 
This means that you have about 4 more decimal places before you start losing pure accuracy.

This will work, because it's within the 64 bits:
```javascript
chance.bool({likelihood: 99.999999998})
```  

and technically this is not guaranteed to be accurate as that number would be truncated to `99.99999999999999999`
```javascript
chance.bool({likelihood: 99.99999999999999998})  // not guaranteed to be accurate
```

https://en.wikipedia.org/wiki/Double-precision_floating-point_format#JavaScript

I hope this resolves your question and you can mark it as answered.
